src/components/TodoModal/TodoModal.tsx

```diff
@@ -1,39 +1,75 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import { Loader } from '../Loader';
+import { getUser } from '../../api';
+import { User } from '../../types/User';
+import { Todo } from '../../types/Todo';
+import classNames from 'classnames';
+
+interface TodoModalProps {
+  todo: Todo;
+  onSetSelectedTodoId: React.Dispatch<React.SetStateAction<number>>;
+}
+
+export const TodoModal: React.FC<TodoModalProps> = ({
+  onSetSelectedTodoId,
+  todo,
+}) => {
+  const [isLoading, setIsLoading] = useState(false);
+  const [user, setUser] = useState<User | null>(null);
+
+  useEffect(() => {
+    setIsLoading(true);
+    getUser(todo.userId)
+      .then(userData => {
+        setUser(userData);
+      })
+      .finally(() => {
+        setIsLoading(false);
+      });
```

Comment:
Ensure that the `getUser` function handles errors gracefully. Consider adding a `.catch()` block to handle any potential errors during the user data fetch, which would improve the robustness of the component.